src/cart/crt_iv.c

```diff
@@ -2550,11 +2550,10 @@ handle_ivupdate_response(const struct crt_cb_info *cb_info)
 		child_output = crt_reply_get(iv_info->uci_child_rpc);
 
 		/* uci_bulk_hdl will not be set for invalidate call */
-		if (iv_info->uci_bulk_hdl != CRT_BULK_NULL) {
+		if (iv_info->uci_bulk_hdl != CRT_BULK_NULL)
 			crt_bulk_free(iv_info->uci_bulk_hdl);
-			iv_ops->ivo_on_put(iv_info->uci_ivns_internal, &iv_info->uci_iv_value,
-					   iv_info->uci_user_priv);
-		}
+		iv_ops->ivo_on_put(iv_info->uci_ivns_internal, &iv_info->uci_iv_value,
```

Comment:
is iv_info->uci_iv_value be valid here if iv_info->uci_bulk_hdl is null? on line 2534 above we do 	d_sgl_buf_copy(&iv_info->uci_iv_value, &output->ivo_iv_sgl);, but this will only happen for bi-directional update.

perhaps we should do line 2534-2535 for all inline cases to populate uci_iv_value at the start

Reply:
As I understand, only bi-directional case needs to transfer back the update value, which needs d_sgl_buf_copy() to copy the data back from Ivo_iv_sgl ?

For other cases, it only needs to free the iv_value allocated before ivu_update_issue(), either it is bulk or inline. Or I miss sth here?

Reply:
ok, i just want to make sure that your implementation of ivo_on_put() will not be doing aything with a value that is not populated, as bulk is null (so either no value, or inline value)